apps/web/src/pages/Communities/CommunityMembersPage/components/AddCommunityMemberDialog.tsx

```diff
@@ -0,0 +1,226 @@
+import {
+  commonMessages,
+  errorMessages,
+  formMessages,
+  getLocalizedName,
+} from "@gc-digital-talent/i18n";
+import {
+  RoleInput,
+  CommunityMembersPage_CommunityFragment as CommunityMembersPageCommunityFragmentType,
+} from "@gc-digital-talent/graphql";
+
+import { getFullNameAndEmailLabel } from "~/utils/nameUtils";
+import { CommunityMember } from "~/utils/communityUtils";
+
+import { CommunityMemberFormValues, ContextType } from "./types";
+import { getTeamBasedRoleOptions } from "./utils";
+import useAvailableUsers from "./useAvailableUsers";
+import useAvailableRoles from "./useAvailableRoles";
+import { UpdateUserCommunityRoles_Mutation } from "./operations";
+
+interface AddCommunityMemberDialogProps {
+  community: CommunityMembersPageCommunityFragmentType;
+  members: Array<CommunityMember>;
+}
+
+const AddCommunityMemberDialog = ({
+  community,
+  members,
+}: // onSave,
+AddCommunityMemberDialogProps) => {
+  const intl = useIntl();
+  const [query, setQuery] = useState<string>("");
+  const {
+    users,
+    total,
+    fetching: usersFetching,
+  } = useAvailableUsers(members, {
+    publicProfileSearch: query || undefined,
+  });
+
+  const { teamId } = useOutletContext<ContextType>();
+  const { roles, fetching: rolesFetching } = useAvailableRoles();
+  const [, executeMutation] = useMutation(UpdateUserCommunityRoles_Mutation);
+  const [isOpen, setIsOpen] = useState<boolean>(false);
+
+  const methods = useForm<CommunityMemberFormValues>({
+    defaultValues: {
+      userId: "",
+      communityId: community.id,
+      roles: [],
+    },
+  });
+
+  const {
+    handleSubmit,
+    formState: { isSubmitting },
+  } = methods;
+
+  const handleSave = async (formValues: CommunityMemberFormValues) => {
+    const roleInputArray: RoleInput[] = formValues.roles.map((role) => {
+      return { roleId: role, teamId };
+    });
+
+    await executeMutation({
+      updateUserRolesInput: {
+        userId: formValues.userId,
+        roleAssignmentsInput: {
+          attach: roleInputArray,
+        },
+      },
+    })
+      .then((res) => {
+        if (!res.error) {
+          setIsOpen(false);
+          toast.success(
+            intl.formatMessage({
+              defaultMessage: "Member added successfully",
+              id: "KESOSJ",
+              description:
+                "Alert displayed to user when a community member was added to a community",
+            }),
+          );
+        }
+      })
+      .catch(() => {
+        toast.error(
+          intl.formatMessage({
+            defaultMessage: "Member addition failed",
+            id: "BbI16K",
+            description:
+              "Alert displayed to user when a community member was not added to a community",
+          }),
+        );
+      });
+  };
+
+  const handleDebouncedSearch = debounce((newQuery: string) => {
+    setQuery(newQuery);
+  }, 300);
+
+  const roleOptions = getTeamBasedRoleOptions(roles, intl);
```

Comment:
I'm not sure the exact way to do this but we should probably filter these roles based on the current users role. When I login as community admin (which can only assign community recruiters) I can see the community admin role even though I cannot actually assign it.

Reply:
d619b1a0c305ce96eb85e4c14c27ebc1836ccadd